### PR TITLE
[server] fix login

### PR DESCRIPTION
**Pull request by AlexTugarev**
Signed-off-by: Alex Tugarev <alex.tugarev@typefox.io>

### CODE REVIEW

**Review by geropl — APPROVED** [review]
:rocket: :+1:

File: components/gitpod-db/src/typeorm/migration/1607429613319-TermsAcceptance-lastModified.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
+import { tableExists } from "./helper/helper";
+
+export class Foo1607429613319 implements MigrationInterface {
+
+    public async up(queryRunner: QueryRunner): Promise<any> {
+        if (!(await tableExists(queryRunner, "d_b_terms_acceptance_entry"))) {
```
Comment (geropl) [inline components/gitpod-db/src/typeorm/migration/1607429613319-TermsAcceptance-lastModified.ts:7]: `if not table exists ALTER COLUMN`? Did you mean `if column not exists`?

File: components/server/src/user/user-controller.ts [hunk 2]
```diff
@@ -249,32 +249,39 @@ export class UserController {
...
             const logContext = LogContext.from({ user: req.user, request: req });
-            const logPayload = { session: req.session, clientInfo, tosFlowInfo, authBag: authFlow };
+            const logPayload = { session: req.session, clientInfo, tosFlowInfo, authFlow, mode };
 
             const redirectOnInvalidSession = () => {
-                log.info(logContext, '(TOS) Invalid session. (/tos)', logPayload);
-                res.redirect(this.getSorryUrl("Oops! Something went wrong in the previous step."));
+                // just don't forget
+                this.tosCookie.unset(res);
+                AuthFlow.clear(req.session);
```
Comment (geropl) [inline components/server/src/user/user-controller.ts:264]: async

File: components/server/src/user/user-controller.ts [hunk 3]
```diff
@@ -322,19 +331,21 @@ export class UserController {
...
+            const tosFlowInfo = TosFlow.get(req.session);
             const authFlow = AuthFlow.get(req.session);
             const isInLoginProcess = !!authFlow;
 
             const logContext = LogContext.from({ user: req.user, request: req });
-            const logPayload = { session: req.session, clientInfo, tosFlowInfo, authBag: authFlow };
+            const logPayload = { session: req.session, clientInfo, tosFlowInfo, authFlow };
 
             const redirectOnInvalidSession = () => {
+                AuthFlow.clear(req.session);
```
Comment (geropl) [inline components/server/src/user/user-controller.ts:342]: async

**Review by geropl — APPROVED** [review]
Nice! :+1: 
(sans `async` comments)

**Comment by AlexTugarev** [comment]
just quickly some-tested the most recent update. merging now.

**Pull request merged**